include/micm/util/vector_matrix.hpp

```diff
@@ -173,7 +218,7 @@ namespace micm
                 {
                   if (other_row.size() != y_dim)
                   {
-                    throw std::runtime_error("Invalid vector for matrix assignment");
+                    throw std::system_error(make_error_code(MicmVectorMatrixErrc::InvalidVector), "");
```

Comment:
Might be worth having a message why it's invalid?

Reply:
done!